sdk/lib/opentelemetry/sdk/resources/auto_detector.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+# Copyright 2019 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+require 'opentelemetry/sdk/resources/detectors/google_cloud_platform'
+require 'opentelemetry/sdk/resources/detectors/telemetry'
+
+module OpenTelemetry
+  module SDK
+    module Resources
+      # AutoDetector contains detect class method for running all detectors
+      module AutoDetector
+        extend self
+
+        DETECTORS = [
```

Comment:
Do we intend to add anything else here? The JS implementation has an AWS detector and environment variable detector (OTEL_RESOURCE_LABELS).

Reply:
For the env detector it wasn't completely clear to me what the behaviour should be there.  I saw this https://github.com/open-telemetry/opentelemetry-specification/issues/572 but I wasn't sure if any standardized convention was set yet.  With a little direction I'll gladly add it.

Regarding the AWS detector, I don't have access to an AWS instance readily available so I didn't want to put out something that might not work in a real world environment.  For example with the GCP detector I also validated it against a production environment.

Reply:
I think people are going to want one for AWS one way or another. We could approach this in one of two ways. We could make an issue for it and let someone pick it up if they want it. Optionally, we could start the process by creating one using the sample output here: https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instance-identity-documents.html.

Reply:
I think we should get the general framework in place with the GCP detector in this PR, and open issues for the other detectors.